packages/access/lib/grants.js

```diff
@@ -183,13 +195,56 @@ const invalidate = async (grant, reason, pgdb, mail) => {
   return result > 0
 }
 
+const followUp = async (campaign, grant, t, pgdb, mail) => {
+  const result = await pgdb.public.accessGrants.update(
+    { id: grant.id, followupAt: null },
+    { followupAt: moment(), updatedAt: moment() }
+  )
+
+  const recipient =
+    await pgdb.public.users.findOne({ id: grant.recipientUserId })
+
+  if (recipient) {
+    const hasMembership =
+      await membershipsLib.hasUserActiveMembership(recipient, pgdb)
+
+    if (!hasMembership) {
+      const grantee = await pgdb.public.users
+        .findOne({ id: grant.granteeUserId })
+
+      await mailLib.sendRecipientFollowup(
+        grantee, campaign, recipient, grant, t, pgdb
+      )
+    }
+  }
+
+  debug('followUp', {
+    id: grant.id,
+    hasRecipient: !!grant.recipientUserId,
+    followUp: moment(),
+    updatedAt: moment(),
```

Comment:
between this debug and the db call on line 199 potentially 4 async calls, why not use the real dates here?

Reply:
Will do.